ape_starknet/ecosystems.py

```diff
@@ -180,7 +182,7 @@ def encode_calldata(
             index += 1
 
         calldata, _ = call_serializer.from_python(*pre_encoded_args)
-        return calldata
+        return List(calldata)
```

Comment:
You have to use lower-case list; the upper case one is only the type declaration.

```suggestion
        return list(calldata)
```